models/event.py

```diff
@@ -24,29 +29,106 @@
 
 
 class FlowModel(models.Model):
-  id = models.ForeignKey('auth.User', primary_key=True)
+  id = models.OneToOneField('auth.User', primary_key=True)
   flow = FlowField()
   class Meta:
     app_label = 'tracker'
 
 
 class CredentialsModel(models.Model):
-  id = models.ForeignKey('auth.User', primary_key=True)
+  id = models.OneToOneField('auth.User', primary_key=True)
   credentials = CredentialsField()
   class Meta:
     app_label = 'tracker'
 
 
+class TimestampValidator(validators.RegexValidator):
+  regex = r'(?:(?:(\d+):)?(?:(\d+):))?(\d+)(?:\.(\d{1,3}))?$'
+  def __call__(self, value):
+    super(TimestampValidator, self).__call__(value)
+    h,m,s,ms = re.match(self.regex, unicode(value)).groups()
+    if h is not None and int(m) >= 60:
+      raise ValidationError('Minutes cannot be 60 or higher if the hour part is specified')
+    if m is not None and int(s) >= 60:
+      raise ValidationError('Seconds cannot be 60 or higher if the minute part is specified')
+
+
+class TimestampField(models.Field):
+  __metaclass__ = models.SubfieldBase
+  default_validators = [TimestampValidator()]
+  match_string = re.compile(r'(?:(?:(\d+):)?(?:(\d+):))?(\d+)(?:\.(\d+))?')
+
+  def __init__(self, always_show_h=False, always_show_m=False, always_show_ms=False, *args, **kwargs):
+    super(TimestampField, self).__init__(*args, **kwargs)
+    self.always_show_h = always_show_h
+    self.always_show_m = always_show_m
+    self.always_show_ms = always_show_ms
+  def to_python(self, value):
```

Comment:
I had to wrestle with this for a while to get it to work, basically it takes either an integer (representing milliseconds) or a string timestamp in the form of [[HH:]MM:]SS[.mmm] and massages it into either something to display, or something to store in the database as a flat integer.

Unfortunately a timedelta doesn't really serve either purpose. It's pretty easy to get one by saying timedelta(milliseconds=*).